xonsh/tools.py

```diff
@@ -2018,11 +2018,14 @@ def format_color(string, **kwargs):
         return ansi_partial_color_format(string, style=style)
 
 
-def print_color(string, **kwargs):
+def print_color(string, color=None, **kwargs):
     """Prints a string that may contain colors. This dispatched to the shell
     method of the same name. Colors will be formatted if they have not already
     been.
     """
+    if color:
+        string = f"{{{color}}}{string}{{RESET}}"
```

Comment:
Nitpick: If it were me, I'd probably write this logic in an `colorize(string, color)` function so that (a) you don't end up mutating the value of `string` conditionally, (b) you can colorize in a single expression instead of a conditional block, and (c) you can test the behavior in isolation.

```python
def colorize(string: str, color: str | None) -> str:
    """
    >>> colorize('input', 'RED')
    '{RED}input{RESET}'
    >>> colorize('input', None)
    'input'
    """
    return f"{{{color}}}{string}{{RESET}}" if color is not None else string
```

I'm not sure if doctests are enabled on the project. If not, you'd probably want to test the behavior in the test suite. I'm unsure if the expected format is `{{RED}}` or `{RED}`.